### PR TITLE
Draft of 0.7.0 release notes

### PR DESCRIPTION
**Pull request by travisbrown**
~~I still think we should include an update guide, but this is a first stab at #1253.~~

This is ready to be reviewed, apart from the fact that ~~#1264, #1265,~~ ~~#1266~~ (saving for next release), ~~and #1267~~ aren't yet included.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,3 +1,112 @@
...
+* [#1193](https://github.com/typelevel/cats/pull/1193): `reduceLeftM` for `Reducible`
+* [#1097](https://github.com/typelevel/cats/pull/1097): Functor variance helpers (`widen` for `Functor` and `narrow` for `Contravariant`)
+* [#1207](https://github.com/typelevel/cats/pull/1207): `tell` for `Writer` and `WriterT`, `value` for `Writer`
+* [#1155](https://github.com/typelevel/cats/pull/1155): Convenience methods for constructing `XorT` values
+* [#1085](https://github.com/typelevel/cats/pull/1085): `runTailRec` and `foldLeftM` for `Free`
+
+And some name changes:
+
+* [#1140](https://github.com/typelevel/cats/pull/1140): `cats.std` is now `cats.instances`
+* [#1201](https://github.com/typelevel/cats/pull/1201): Many `*Unsafe` methods are now `unsafe*`
```
Comment (fthomas) [inline CHANGES.md]: This PR wasn't merged.

Comment (travisbrown) [inline CHANGES.md]: Ah, thanks @fthomas. There's a possibility there's also stuff still missing—I'll do a closer pass this afternoon.

**Comment by travisbrown** [comment]
I went ahead and added commits adding new contributors and updating versions as described in the [process](https://github.com/typelevel/cats/blob/master/PROCESS.md) document.

**Comment by kailuowang** [comment]
#1264 was merged. It's not clear to me why #1265 and #1267 are not merged yet (seems they both got enough sign-offs).

**Comment by travisbrown** [comment]
@kailuowang Yeah, I wanted to wait for all three `NonEmpty*` PRs to land before updating the release notes.

File: CHANGES.md [hunk 2]
```diff
@@ -1,3 +1,128 @@
+## Version 0.7.0
+
+> 2016 August 2
+
+Version 0.7.0 is the seventh Cats release, and includes several major rearrangements and changes to names.
+
+### Migration notes
```
Comment (ceedubs) [inline CHANGES.md]: Thank you for adding this section @travisbrown. I think this will be really helpful for people.

**Comment by johnynek** [comment]
I'd love to finish #1280 before 0.7.0. It would be a shame to have 0.7.0 introduce FlatMapRec only to remove it in the next release.

I can get 1280 passing in a day or two if we can wait.

File: CHANGES.md [hunk 3]
```diff
@@ -1,3 +1,132 @@
...
+* `NonEmptyList` and `NonEmptyVector` are no longer type aliases for `OneAnd`, so any code using `OneAnd` to construct or pattern match on these types will need to be changed to use `NonEmptyList` or `NonEmptyVector` directly. There are also some API changes; for example, `unwrap` calls will need to be replaced by `toList` or `toVector`, and `NonEmptyList(1, 2, 3)` is now `NonEmptyList.of(1, 2, 3)`.
+* `pureEval` has been removed from `Applicative` ([#1234](https://github.com/typelevel/cats/pull/1234)), and has not been replaced, so if you are relying on it for laziness or effect capturing (which wasn't enforced or guaranteed), you'll need to find another approach.
```
Comment (johnynek) [inline CHANGES.md]: What I suggested was to require a `NaturalTransformation[Eval, M]` if you need to create an `M[T]` from an `Eval[T]`. For `M[T]` with laziness, you can do something non-trivial.

**Comment by non** [comment]
LGTM 👍

**Comment by non** [comment]
Since this is documentation I'm going to merge this after one +1.

**Pull request merged**